python/ml4ir/applications/ranking/features/feature_fns/string.py

```diff
@@ -64,8 +69,15 @@ def call(self, inputs, training=None):
         else:
             query_len = tf.strings.length(inputs)
 
-        query_len = tf.expand_dims(tf.cast(query_len, tf.float32), axis=-1)
-        return query_len
+        if self.one_hot:
+            # Clip the query lengths to the max_length
+            query_len = tf.clip_by_value(query_len, 0, self.max_length)
+            # Convert to one-hot encoding
+            query_len_one_hot = tf.one_hot(query_len, depth=self.max_length + 1)
```

Comment:
@mohazahran there's another function in ml4ir I believe which does the one hot, but this should be fine.